### PR TITLE
Add tests for IMVJ with restart methods

### PR DESCRIPTION
**Pull request by Fujikawas**
## Main changes of this PR
Added integration tests which employed IMVJ acceleration method with various restart methods.

## Motivation and additional information
To solve the low coverage problem described in https://github.com/precice/precice/issues/1463

## Author's checklist

* [x] I used the [`pre-commit` hook](https://precice.org/dev-docs-dev-tooling.html#setting-up-pre-commit) to prevent dirty commits and used `pre-commit run --all` to format old commits.
* [x] I squashed / am about to squash all commits that should be seen as one.

## Reviewers' checklist

* [x] Do you understand the code changes?

### CODE REVIEW

**Comment by Fujikawas** [comment]
Current `IQNIMVJAcceleration.cpp`:  `Coverage% = 86%`

File: tests/quasi-newton/parallel/TestQN6.cpp [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+#include <precice/precice.hpp>
+#include "../helpers.hpp"
+
+BOOST_AUTO_TEST_SUITE(Integration)
+BOOST_AUTO_TEST_SUITE(QuasiNewton)
+BOOST_AUTO_TEST_SUITE(Parallel)
+BOOST_AUTO_TEST_CASE(TestQN6)
+{
+  PRECICE_TEST("SolverOne"_on(2_ranks), "SolverTwo"_on(2_ranks));
+  // serial coupling, IQN-IMVJ (which is identical to IQN-ILS as only first timestep is considered), relaxed QR2 filter
```
Comment (uekerman) [inline tests/quasi-newton/parallel/TestQN6.cpp:14]: Is this comment actually correct? The config looks like this test does 5 time windows?

**Review by uekerman — APPROVED** [review]
Very good additions :+1:

Unit tests that actually test whether we do the correct thing could also be helpful, but that's for another PR.

@Fujikawas If you have a good idea, we could try renaming "QN7" etc to sth more readable. But not a must.

File: tests/quasi-newton/parallel/TestQN6.cpp (hunk 1 above)
Comment (Fujikawas) [reply tests/quasi-newton/parallel/TestQN6.cpp:14]: Yes, it's 5 time windows. I'll improve them. 
And I would like to add the goals of relevant tests here in the comments so we don't need to do a lot of renaming and still provide some insight.

**Pull request merged**